File: sdks/packages/midan/src/events-types/update-event-team-member-event.ts
import { PublicKey } from '@solana/web3.js';
import { BN } from '@coral-xyz/anchor';
import {EventTeamMemberStatusType} from "../types/midan-enums";

export const UpdateEventTeamMemberEventName = "UpdateEventTeamMemberEvent";

export interface UpdateEventTeamMemberEvent {
    timestamp: BN,

    creatorKey: PublicKey,

    authority: PublicKey,

    member: PublicKey,

    eventTeamIndex: number,

    eventTeamMemberStatus: EventTeamMemberStatusType
}

File: programs/midan/src/events/update_event_team_member_event.rs
use anchor_lang::prelude::*;
use crate::states::EventTeamMemberStatus;

#[event]
pub struct UpdateEventTeamMemberEvent {
    pub timestamp: i64,

    pub creator_key: Pubkey,

    pub authority: Pubkey,
    
    pub member: Pubkey,

    pub event_team_index: u32,
    
    pub event_team_member_status: EventTeamMemberStatus,
}